src/condition/lfi_detector.cpp

```diff
@@ -17,125 +16,116 @@ namespace ddwaf {
-                if (path.find(key) != std::string_view::npos) {
-                    strings.emplace_back(key);
-                }
-            }
-
-            if (object::is_container(&child)) {
-                if (stack.size() < object_limits::default_max_container_depth) {
-                    stack.push(&child);
-                }
-                continue;
-            }
-
-            if (child.type == DDWAF_OBJ_STRING && child.nbEntries >= min_str_len) {
-                const std::string_view value{
-                    child.stringValue, static_cast<std::size_t>(child.nbEntries)};
-                if (path.find(value) != std::string_view::npos) {
-                    strings.emplace_back(value);
-                }
-            }
-        }
-    }
+using lfi_result = std::optional<std::pair<std::string, std::vector<std::string>>>;
 
-    return strings;
-}
-
-std::vector<std::string_view> split(std::string_view str, char sep)
+bool find_directory_escape(std::string_view value, std::string_view sep)
 {
-    std::vector<std::string_view> components;
-
     std::size_t start = 0;
-    while (start < str.size()) {
-        const std::size_t end = str.find(sep, start);
+    unsigned part_count = 0;
+    bool part_seen = false;
+    while (start < value.size()) {
+        const std::size_t end = value.find_first_of(sep, start);
 
         if (end == start) {
             // Ignore zero-sized strings
             start = end + 1;
+            continue;
         }
 
-        if (end == std::string_view::npos) {
-            // Last element
-            components.emplace_back(str.substr(start));
-            start = str.size();
-        } else {
-            components.emplace_back(str.substr(start, end - start));
+        ++part_count;
+
+        std::string_view part;
+        if (end != npos) {
+            part = value.substr(start, end - start);
             start = end + 1;
+        } else {
+            part = value.substr(start);
+            start = value.size();
+        }
+
+        part_seen = part_seen || part == "..";
+        if (part_count > 1 && part_seen) {
+            return true;
         }
     }
 
-    return components;
+    return false;
 }
 
-std::pair<bool, std::string> lfi_impl(std::string_view path, const ddwaf_object &params)
+// TODO: https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats#dos-device-paths
+bool lfi_impl_windows(std::string_view path, std::string_view param)
 {
-    auto matched_params = match_params(path, params);
-    if (matched_params.empty()) {
-        return {};
+    static constexpr std::size_t min_str_len = 2;
+
+    if (param.size() < min_str_len && !path.ends_with(param)) {
+        return false;
     }
 
-    for (const auto &param : matched_params) {
-        if (param[0] == '/' && param == path) {
-            return {true, std::string(param)};
+    bool is_absolute =
+        param[0] == '/' || param[0] == '\\' || (ddwaf::isalpha(param[0]) && param[1] == ':');
```

Comment:
Should be `bool is_absolute =
        param[0] == '/' || param[0] == '\\' || param.size() >= 3 &&(ddwaf::isalpha(param[0]) && param[1] == ':' && (param[2] == '/' || param[2] == '\\')));